### PR TITLE
EZP-32186: FormMapper for Datatype uses obscure syntax

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/EZP-32186
| Versions      | 3.x, master

Note: This reuses code from the FT tutorial, instead of basing on the built-in FTs

### CODE REVIEW

**Review by vidarl — APPROVED** [review]
This looks good to me

But did you double-check with someone in engineering if they find it valuable to use the formFactory in the example?
My opinion is stated in https://issues.ibexa.co/browse/EZP-32186, someone might not agree

File: docs/api/field_type_form_and_template.md [hunk 1]
```diff
@@ -17,39 +17,17 @@ The `FieldValueFormMapperInterface::mapFieldValueForm` method accepts two argume
...
-}
+[[= include_file('code_samples/field_types/2dpoint_ft/src/FieldType/Point2D/Type.php', 6, 7) =]][[= include_file('code_samples/field_types/2dpoint_ft/src/FieldType/Point2D/Type.php', 10, 11) =]][[= include_file('code_samples/field_types/2dpoint_ft/src/FieldType/Point2D/Type.php', 12, 13) =]]
+// ...
+
+[[= include_file('code_samples/field_types/2dpoint_ft/src/FieldType/Point2D/Type.php', 33, 41) =]]
 ```
 
 Your type has to be called `value`.
-In the example above, `CheckboxFieldType::class` is used, but you can use standard Symfony form type instead.
+In the example above, the custom `Point2DType::class` is used, but you can use standard Symfony form type instead.
```
Comment (dabrt) [inline docs/api/field_type_form_and_template.md:30]:
```suggestion
In the example above, the custom `Point2DType::class` is used, but you can use a standard Symfony form type instead.
```
Unless there is only one (seems like not), in wchich case the article would be "the".

File: docs/api/field_type_type_and_value.md [hunk 2]
```diff
@@ -143,24 +140,15 @@ This ensures that the initialization steps shared by all Field Types are execute
 
 It is recommended to use a simple associative array format for the settings schema returned by `eZ\Publish\SPI\FieldType\FieldType::getSettingsSchema()`, which follows these rules:
 
-- The key of the associative array identifies a setting (e.g. `default`)
+- The key of the associative array identifies a setting (e.g. `format`)
```
Comment (dabrt) [inline docs/api/field_type_type_and_value.md:143]:
```suggestion
- The key of the associative array identifies a setting (for example, `format`)
```

File: docs/api/field_type_type_and_value.md [hunk 3]
```diff
@@ -143,24 +140,15 @@ This ensures that the initialization steps shared by all Field Types are execute
...
 - Its value is an associative array describing the setting using:
     - `type` to identify the setting type (e.g. `int` or `string`)
```
Comment (dabrt) [inline docs/api/field_type_type_and_value.md:145]:
```suggestion
    - `type` to identify the setting type (for example, `int` or `string`)
```

**Review by dabrt — APPROVED** [review]

**Comment by DominikaK** [comment]
After verification with engineering, the current code is correct, what we need to do is remove the confusing comment in code, as it refers to an internal DataTransformer (will be done separately).

**Pull request closed without merging**